filament/src/materials/MaterialHelpersShading.fxh

```diff
@@ -0,0 +1,389 @@
+    vec2 uvY = scaler * pos.xz; // y facing plane
+    vec2 uvZ = scaler * pos.xy; // z facing plane
+
+    // Tangent space normal maps
+    // 2-channel XY TS normal texture: this saves 33% on storage
+    lowp vec3 tnormalX = UnpackNormal(texture(normalMap, uvX).xy);
+    lowp vec3 tnormalY = UnpackNormal(texture(normalMap, uvY).xy);
+    lowp vec3 tnormalZ = UnpackNormal(texture(normalMap, uvZ).xy);
+
+    // Swizzle world normals into tangent space and apply Whiteout blend
+    tnormalX = vec3(tnormalX.xy + normal.zy, abs(tnormalX.z) * normal.x);
+    tnormalY = vec3(tnormalY.xy + normal.xz, abs(tnormalY.z) * normal.y);
+    tnormalZ = vec3(tnormalZ.xy + normal.xy, abs(tnormalZ.z) * normal.z);
+    // Compute blend weights
+    vec3 blend = ComputeWeights(normal);
+    // Swizzle tangent normals to match world orientation and triblend
+    return normalize(tnormalX.zyx * blend.x + tnormalY.xzy * blend.y + tnormalZ.xyz * blend.z);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//
+// Custom code to evaluate the materials
+//
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void ApplyNormalMap(inout MaterialInputs material, inout FragmentData fragmentData) {
+#if defined(MATERIAL_HAS_NORMAL)
+    if (materialParams.useNormalTexture == 1) {
+        // We combine the normals in world space, hence the transformation in the end from world to tangent, assuming an
+        // orthonormal tangent frame (which may not hold actually but looks fine enough for now).
+        vec3 normalWS = TriplanarNormalMap(materialParams_normalTexture,
+                                           materialParams.textureScaler.y,
+                                           fragmentData.pos,
+                                           fragmentData.normal);
+        material.normal = normalWS * getWorldTangentFrame();
+    }
+    // By this time the normal should be ready, it is safe to apply the normal scale
+    material.normal.xy *= materialParams.normalIntensity;
+#endif
+}
+
+void ApplyClearCoatNormalMap(inout MaterialInputs material, inout FragmentData fragmentData) {
+#if defined(MATERIAL_HAS_CLEAR_COAT_NORMAL)
+    if (materialParams.useClearCoatNormalTexture == 1) {
+        vec3 clearCoatNormalWS = TriplanarNormalMap(materialParams_clearCoatNormalTexture,
+                                                    materialParams.textureScaler.z,
+                                                    fragmentData.pos,
+                                                    fragmentData.normal);
+        material.clearCoatNormal = clearCoatNormalWS * getWorldTangentFrame();
+        material.clearCoatNormal.xy *= materialParams.clearCoatNormalIntensity;
+    }
+#endif
+}
+
+void ApplyBaseColor(inout MaterialInputs material, inout FragmentData fragmentData) {
+#if defined(MATERIAL_HAS_BASE_COLOR) 
```

Comment:
👾 
```suggestion
#if defined(MATERIAL_HAS_BASE_COLOR)
```

Reply:
Fixed in https://github.com/shapr3d/filament/pull/27/commits/f17aa20c12b5006d3662f3b03c47ac862af71a0d